src/content/docs/de/guides/client-side-scripts.mdx

```diff
@@ -0,0 +1,237 @@
+
+```astro title="src/components/AstroHeart.astro"
+<!-- Schließe die Komponentenelemente in unser benutzerdefiniertes Element "astro-heart" ein. -->
+<astro-heart>
+  <button aria-label="Heart">💜</button> × <span>0</span>
+</astro-heart>
+
+<script>
+  // Definiere das Verhalten für unseren neuen Typ von HTML-Element.
+  class AstroHeart extends HTMLElement {
+    constructor() {
+			super();
+      let count = 0;
+
+      const heartButton = this.querySelector('button');
+      const countSpan = this.querySelector('span');
+
+      // Jedes Mal, wenn die Schaltfläche angeklickt wird, aktualisiere die Anzahl.
+			heartButton.addEventListener('click', () => {
+        count++;
+        countSpan.textContent = count;
+      });
+		}
+  }
+
+  // Teile dem Browser mit, dass er unsere AstroHeart-Klasse für <astro-heart>-Elemente verwenden soll.
+  customElements.define('astro-heart', AstroHeart);
+</script>
+```
+
+Die Verwendung eines benutzerdefinierten Elements hat hier zwei Vorteile:
+
+1. Anstatt die gesamte Seite mit `document.querySelector()` zu durchsuchen, kannst du `this.querySelector()` verwenden, das nur innerhalb der aktuellen benutzerdefinierten Elementinstanz sucht. Das macht es einfacher, jeweils nur mit den Children einer Komponenteninstanz zu arbeiten.
+
+2. Obwohl ein `<script>` nur einmal ausgeführt wird, führt der Browser die Methode `constructor()` unseres benutzerdefinierten Elements jedes Mal aus, wenn er `<astro-heart>` auf der Seite findet. Das bedeutet, dass du gefahrlos Code für jeweils eine Komponente schreiben kannst, auch wenn du diese Komponente mehrmals auf einer Seite verwenden willst.
+
+📚 Mehr über benutzerdefinierte Elemente erfährst du in [web.dev's Reusable Web Components guide](https://web.dev/custom-elements-v1/) und [MDN's introduction to custom elements](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements).
+
+### Übergabe von Frontmatter-Variablen an Skripte
+
+In Astro-Komponenten läuft der Code in [Frontmatter](/de/core-concepts/astro-components/#das-komponentenskript) zwischen den `---`-Zeichen auf dem Server und ist im Browser nicht verfügbar. Um Variablen vom Server an den Client zu senden, brauchen wir eine Möglichkeit, unsere Variablen zu speichern und sie dann zu lesen, wenn JavaScript im Browser läuft.
+
+Eine Möglichkeit, dies zu tun, ist die Verwendung von [`data-*`-Attributen](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes), um den Wert von Variablen in deiner HTML-Ausgabe zu speichern. Skripte, einschließlich benutzerdefinierter Elemente, können diese Attribute dann mit der Eigenschaft `dataset` eines Elements lesen, sobald dein HTML im Browser geladen ist.
+
+In dieser Beispielkomponente wird eine "Message"-Eigenschaft in einem "Data-Message"-Attribut gespeichert, sodass das benutzerdefinierte Element "this.dataset.message" lesen und den Wert der Eigenschaft im Browser abrufen kann.
```

Comment:
Similar issue, with words that should've been wrapped in backticks. Not sure if this is something particular to the German language, so feel free to ignore this if that's the case.
```suggestion
In dieser Beispielkomponente wird eine `message`-Eigenschaft in einem `data-message`-Attribut gespeichert, sodass das benutzerdefinierte Element `this.dataset.message` lesen und den Wert der Eigenschaft im Browser abrufen kann.
```